Index: gtk+2.0-2.21.5/gtk/updateiconcache.c
===================================================================
--- gtk+2.0-2.21.5.orig/gtk/updateiconcache.c	2010-05-08 06:39:07.000000000 +0200
+++ gtk+2.0-2.21.5/gtk/updateiconcache.c	2010-07-22 17:57:21.973055310 +0200
@@ -660,7 +660,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 		
 	      image = g_new0 (Image, 1);
