Index: gtk+2.0-2.21.5/modules/printbackends/cups/gtkprintbackendcups.c
===================================================================
--- gtk+2.0-2.21.5.orig/modules/printbackends/cups/gtkprintbackendcups.c	2010-06-28 01:19:39.000000000 +0200
+++ gtk+2.0-2.21.5/modules/printbackends/cups/gtkprintbackendcups.c	2010-07-22 17:57:22.973051749 +0200
@@ -344,8 +344,9 @@
  
   /* TODO: check if it is a ps or pdf printer */
   
-  surface = cairo_ps_surface_create_for_stream  (_cairo_write_to_cups, cache_io, width, height);
+  surface = cairo_pdf_surface_create_for_stream  (_cairo_write_to_cups, cache_io, width, height);
 
+  /*
   ppd_file = gtk_printer_cups_get_ppd (GTK_PRINTER_CUPS (printer));
 
   if (ppd_file != NULL)
@@ -403,6 +404,7 @@
 
   if (level == 3)
     cairo_ps_surface_restrict_to_level (surface, CAIRO_PS_LEVEL_3);
+  */
 
   cairo_surface_set_fallback_resolution (surface,
                                          2.0 * gtk_print_settings_get_printer_lpi (settings),
Index: gtk+2.0-2.21.5/modules/printbackends/lpr/gtkprintbackendlpr.c
===================================================================
--- gtk+2.0-2.21.5.orig/modules/printbackends/lpr/gtkprintbackendlpr.c	2010-06-16 17:27:06.000000000 +0200
+++ gtk+2.0-2.21.5/modules/printbackends/lpr/gtkprintbackendlpr.c	2010-07-22 17:57:23.001065368 +0200
@@ -31,6 +31,8 @@
 #include <cairo.h>
 #include <cairo-ps.h>
 
+#include <cairo-pdf.h>
+
 #include <glib/gi18n-lib.h>
 
 #include <gtk/gtk.h>
@@ -209,7 +211,7 @@
 {
   cairo_surface_t *surface;
   
-  surface = cairo_ps_surface_create_for_stream (_cairo_write, cache_io, width, height);
+  surface = cairo_pdf_surface_create_for_stream (_cairo_write, cache_io, width, height);
 
   cairo_surface_set_fallback_resolution (surface,
                                          2.0 * gtk_print_settings_get_printer_lpi (settings),
